addons/binding/org.openhab.binding.ikeatradfri/src/main/java/org/openhab/binding/ikeatradfri/handler/IkeaTradfriGatewayHandler.java

```diff
@@ -0,0 +1,339 @@
+                    logger.debug("COAP PUT Error");
+                    future.completeExceptionally(null);
+                    removeAsyncClient(client);
+                }
+            };
+            addAsyncClient(client);
+            client.put(handler, payload, MediaTypeRegistry.TEXT_PLAIN);
+            return future;
+        } catch (URISyntaxException e) {
+            logger.warn("COAP URI exception: {}", e.getMessage());
+            future.completeExceptionally(null);
+        }
+        return future;
+    }
+
+    private void addAsyncClient(CoapClient client) {
+        asyncClients.add(client);
+    }
+
+    private void removeAsyncClient(CoapClient client) {
+        client.shutdown();
+        asyncClients.remove(client);
+    }
+
+    @Override
+    public void onDataUpdate(JsonElement json) {
+        logger.debug("Observation data: {}", json.toString());
+
+        try {
+            JsonArray array = json.getAsJsonArray();
+            for (int i = 0; i < array.size(); i++) {
+                deviceDiscoverHelper(array.get(i).getAsString());
+            }
+        } catch (JsonSyntaxException e) {
+            logger.warn("JSON error: {}", e.getMessage());
+        }
+
+        for (ThingUID thingUID : pendingObserve) {
+            Thing thing = getThingByUID(thingUID);
+            if (thing.getHandler() != null) {
+                observeDevice(thingUID, (IkeaTradfriBulbHandler) thing.getHandler());
+            }
+        }
+        pendingObserve.clear();
+    }
+
+    private void deviceDiscoverHelper(String deviceId) {
+        coapGET("15001/" + deviceId).thenAccept(data -> {
+            logger.debug("Got response {}\nListeners {}", data, dataListeners.size());
+            // Trigger a new discovery of things
+            JsonObject json2 = new JsonParser().parse(data).getAsJsonObject();
+            for (IkeaTradfriDiscoverListener dataListener : dataListeners) {
+                dataListener.onDeviceFound(getThing().getUID(), json2);
+            }
+        });
+    }
+
+    private void observe(String url, ThingUID thingUID, IkeaTradfriObserveListener listener) {
+        logger.debug("Observing: {}", url);
+        IkeaTradfriGatewayConfiguration configuration = getConfigAs(IkeaTradfriGatewayConfiguration.class);
+        try {
+            URI uri = new URI("coaps://" + configuration.host + "//" + url);
+
+            CoapClient client = new CoapClient(uri);
+            client.setEndpoint(endPoint);
+            CoapHandler handler = new CoapHandler() {
+                @Override
+                public void onLoad(CoapResponse response) {
+                    logger.debug("COAP Observe: \noptions: {}\npayload: {} ", response.getOptions().toString(),
+                            response.getResponseText());
+                    if (response.isSuccess()) {
+                        try {
+                            listener.onDataUpdate(parser.parse(response.getResponseText()));
```

Comment:
Funny observation: When I change the valued from the Trådfri App, then this here gets notified on the _first_ change. Afterwards it looses the contact.

I have no clue about the californium API, but from wildly guessing adding this line appears to fix it:

```
observeRelationMap.get(thingUID).reregister();
```

So it looks like such an observation relation needs to be re-registered once it fired. But as I said, I actually have no clue...